src/NUnitTestAdapter/NUnitEngine/NUnitTestEvent.cs

```diff
@@ -199,8 +189,25 @@ public NUnitTestEventTestCase(XmlNode node) : base(node)
 
         public string ReasonMessage { get; }
 
-        public bool HasReason => string.IsNullOrEmpty(ReasonMessage);
+        public bool HasReason => !string.IsNullOrEmpty(ReasonMessage);
         public bool HasFailure => Failure != null;
+
+        /// <summary>
+        /// Find stacktrace in assertion nodes if not defined (seems .netcore2.0 doesn't provide stack-trace for Assert.Fail("abc"))
```

Comment:
> seems .netcore2.0 doesn't provide stack-trace for Assert.Fail("abc")

This sounds like a potentially fixable framework bug.

Reply:
Should it be raised in the NUnit repo?

Reply:
Yes please, if you know how to reproduce it.

Reply:
I can't repro with netcore 2.1.  And, with net core 2.0 I can't even execute the tests:  1>D:\repos\NUnit\FrameworkIssues\NetCore20NoStackTrace\NetCore20NoStackTrace.csproj : warning NU1701: Package 'NUnit3TestAdapter 3.16.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v2.0'. This package may not be fully compatible with your project.

Have we lost the netcore 2.0 compatibility ?

Reply:
If I try with earlier versions (than 3-.16.1 of the adapter) it says:  "Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Unable to find D:\repos\NUnit\FrameworkIssues\NetCore20NoStackTrace\bin\Debug\netcoreapp2.0\testhost.dll. Please publish your test project and retry."

Reply:
.NET Core 2.1 and 3.1 are the oldest two .NET Core *runtime* versions still supported by Microsoft. We only support back to 2.1:

https://github.com/nunit/nunit3-vs-adapter/blob/26aa1f0304a75fe2937cda95b4ebe7eb3ef19888/src/NUnitTestAdapter/NUnit.TestAdapter.csproj#L8

If 2.1 doesn't repro, then I'm not worried about it.

Test projects should have no reason to target 2.0 rather than 2.1 because 2.0 is a deprecated runtime.

Reply:
Ok, the adapter is built with netcore2.1.  I assumed that could run 2.0, but obviously not.  No one has reported anything about this, so should we care?

Reply:
This is the flip side of not having `developmentDependency="true"`. I still think not having it is the right thing to do. It means the target framework and version now must match the project that it's installed into.

I don't think we should care. If someone asks, we say, "Move to 2.1 because not even Microsoft supports 2.0."

Reply:
And trying a .net framework 3.5 gives this: "3>d:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2081,5): warning MSB3268: The primary reference "C:\Users\TerjeSandstrom\.nuget\packages\nunit\3.12.0\lib\net35\nunit.framework.dll" could not be resolved because it has an indirect dependency on the framework assembly "System.Web, "

Reply:
That's because the adapter requires System.Web for one interface which NUnit Engine uses. The answer is to make the test project target the normal full `net35`.

Reply:
Version 3.16 stopped supporting net core 1.0 **and** net core 2.0.    Version 3.15.1 was the latest with support for those. And it needs the Test.Sdk 15, doesn't work with 16.  And - the stacktrace is there... so I think the comment is bogus....or outdated....

Reply:
Cool, then I'll mark this comment thread as resolved.